Code/RDGeneral/CMakeLists.txt

```diff
@@ -6,7 +6,29 @@ rdkit_library(RDGeneral
               LocaleSwitcher.cpp versions.cpp SHARED
               LINK_LIBRARIES ${RDKit_THREAD_LIBS})
 
-target_link_libraries(RDGeneral ${RDKit_THREAD_LIBS})
+if(MINGW)
+  set(hasMSVCRuntime "FALSE")
+  if(RDK_BUILD_THREADSAFE_SSS)
+    set(needMSVCRuntime "TRUE")
+  else()
+    set(needMSVCRuntime "FALSE")
+  endif()
+  if(needMSVCRuntime AND MSVC_RUNTIME_DLL)
+    if(EXISTS "${MSVC_RUNTIME_DLL}")
+      set(hasMSVCRuntime "TRUE")
+    endif()
+  endif()
+  if(needMSVCRuntime AND (NOT hasMSVCRuntime))
+    set(systemRootPath "$ENV{SystemRoot}")
+    string(REGEX REPLACE "\\\\" "/" systemRootPath "${systemRootPath}")
+    message(FATAL_ERROR "Please set MSVC_RUNTIME_DLL to the full path to a msvcrXXX.dll "
+      "Microsoft Visual C Runtime DLL appropriate for your system; the most likely path is "
+      "${systemRootPath}/System32, or ${systemRootPath}/SysWOW64 in case you are building "
+      "32-bit RDKit under a 64-bit Windows OS.")
+  endif()
+endif(MINGW)
+
+target_link_libraries(RDGeneral ${RDKit_THREAD_LIBS} ${MSVC_RUNTIME_DLL})
```

Comment:
The reason the MSVC runtime needs to be linked to RDGeneral is that when multithreaded support is compiled in, LocalSwitcher.cpp needs _configthreadlocale(), which is only available as part of the MSVC runtime:

`#ifdef RDK_THREADSAFE_SSS
      _configthreadlocale(_ENABLE_PER_THREAD_LOCALE);`

So the linker complains about an undefined reference to _configthreadlocale() if the MSVC runtime is not linked to RDGeneral, but that does not affect the other libraries. All C++ tests are passed.